Automattic/wp-calypso
# Receipts: Convert receipt assembler to TypeScript

#### Proposed Changes

This converts the receipt assembler functions to TypeScript. They were already using the `@ts-check` pragma with typedef types so there wasn't much to do. I moved the type definitions to their own file and renamed them slightly to differentiate the term "purchase" for receipts from the already-existing "purchase" type for purchases.

#### Testing Instructions

Verify there are no TypeScript errors reported by the build process.

There was really only one functional change which I've noted inline.
